core/lib/src/fairing/mod.rs

```diff
@@ -153,6 +157,22 @@ pub type Result<T = Rocket<Build>, E = Rocket<Build>> = std::result::Result<T, E
 ///     via response callbacks. Any modifications to a request are persisted and
 ///     can potentially alter how a request is routed.
 ///
+///   * **<a name="request">RequestFilter</a> (`on_request_filter`)**
+///
+///     A request callback, represented by the [`Fairing::on_request_filter()`] method,
+///     is called just after a request is received, immediately after
+///     pre-processing the request and running all `Request` fairings. This method
+///     returns a `Result`, which can be used to terminate processing of a request,
+///     bypassing the routing process. The error value must be a [`TypedError`], which
+///     can then be caught by a typed catcher.
```

Comment:
This is very interesting. I apologize if I missed it, but what's the rationale for this approach as opposed to changing request fairings to optionally return an error and abort the request? It seems desirable to be able to abort a request _before_ other request fairings are run, for example, or to be able to choose the order based on how you attach fairings.